packageurl_test.go

```diff
@@ -119,150 +116,198 @@ func (t TestFixture) Qualifiers() packageurl.Qualifiers {
 	return q
 }
 
-// TestFromStringExamples verifies that parsing example strings produce expected
-// results.
-func TestFromStringExamples(t *testing.T) {
-	// Read the json file
-	data, err := os.ReadFile("testdata/test-suite-data.json")
-	if err != nil {
-		t.Fatal(err)
+type ComponentsOrPurl struct {
+	Purl          *string
+	PurlComponent *ComponentData
+}
+
+func (cop *ComponentsOrPurl) UnmarshalJSON(data []byte) error {
+	// Try string first
+	var s string
+	if err := json.Unmarshal(data, &s); err == nil {
+		cop.Purl = &s
+		return nil
+	}
+
+	var comp ComponentData
+	if err := json.Unmarshal(data, &comp); err == nil {
+		cop.PurlComponent = &comp
+		return nil
 	}
-	// Load the json file contents into a structure
-	testData := []TestFixture{}
-	err = json.Unmarshal(data, &testData)
+
+	return fmt.Errorf("ComponentsOrPurl: data is neither a string nor PURL component")
+}
+
+type TestFixture struct {
+	Description        string           `json:"description"`
+	TestGroup          string           `json:"test_group"`
+	TestType           string           `json:"test_type"`
+	Input              ComponentsOrPurl `json:"input"`
+	ExpectedFailure    bool             `json:"expected_failure"`
+	ExpectedOutput     ComponentsOrPurl `json:"expected_output"`
+	ExpectedFailureMsg *string          `json:"expected_failure_reason"`
```

Comment:
Appears unused at the moment but still better to keep naming aligned.
```suggestion
	ExpectedFailureReason *string          `json:"expected_failure_reason"`
```